packages/backend/src/handlers/actions/player/syncBalances.ts

```diff
@@ -50,86 +59,141 @@ const setBalances = async ({
+            tokenAddress: address,
+            chainId,
+          });
+          const {
+            guild: [{ guild_players: players }],
+          } = await client.GetGuildMembers({ id: guildId });
+          const playerRet = await Promise.all(
+            players.map(
+              async ({
+                Player: { ethereumAddress: ethAddress, id: playerId },
+              }) => {
+                const total = await client.GetTotalForPlayer({
+                  tokenAddress: address,
+                  playerAddress: ethAddress,
+                });
+                const balance =
+                  total.balance_aggregate.aggregate?.sum?.amount ?? 0;
 
-            const seasonalTotal = await client.GetTotalForPlayer({
-              tokenAddress: address,
-              playerAddress: player.Player.ethereumAddress,
-              executedAfter: getCurrentSeasonStart(),
-            });
-            const seasonalBalance =
-              seasonalTotal.balance_aggregate.aggregate?.sum?.amount;
+                const seasonalTotal = await client.GetTotalForPlayer({
+                  tokenAddress: address,
+                  playerAddress: ethAddress,
+                  executedAfter: seasonStart,
+                });
+                const seasonalBalance =
+                  seasonalTotal.balance_aggregate.aggregate?.sum?.amount ?? 0;
 
-            const {
-              xp: [{ initial } = { initial: 0 }],
-            } = await client.GetInitialXP({ playerId: player.Player.id });
+                const {
+                  xp: [{ initial } = { initial: 0 }],
+                } = await client.GetInitialXP({ playerId });
+                const calculated = balance * multiplier + initial;
 
-            await client.UpsertXP({
-              balance: (balance ?? 0) + initial,
-              playerId: player.Player.id,
-              tokenAddress: address,
-              seasonalBalance,
-            });
-          }),
-        );
-      },
-    ),
-  );
-  const ranks = await client.GetPlayersByTotalXP();
+                await client.UpsertXP({
+                  balance: calculated,
+                  playerId,
+                  tokenAddress: address,
+                  seasonalBalance,
+                });
 
-  Promise.allSettled(
-    ranks.xp.map(async ({ playerId, seasonalBalance, balance }, index) => {
-      const rank = computeRank(index);
-      await client.UpdateProfileXP({
-        playerId,
-        seasonXP: seasonalBalance,
-        totalXP: balance,
-        rank,
-      });
-    }),
-  );
-  res.json('Complete! XP saved');
+                return {
+                  address: ethAddress,
+                  xp: {
+                    initial,
+                    accumulated: balance,
+                    calculated,
+                    seasonal: seasonalBalance,
+                  },
+                };
+              },
+            ),
+          );
+          return {
+            ...balRet,
```

Comment:
can we use full words for return, balanceReturned or something. I have context and still confusing